firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -84,4 +248,60 @@ public boolean isTesterSignedIn() {
 
   /** Signs out the App Distribution tester */
   public void signOutTester() {}
+
+  @Override
+  public void onActivityCreated(
+      @NonNull @NotNull Activity activity, @androidx.annotation.Nullable @Nullable Bundle bundle) {
+    Log.d(TAG, "Created activity: " + activity.getClass().getName());
+    // if signinactivity is created, sign-in was succesful
+    if (currentlySigningIn && activity instanceof SignInResultActivity) {
+      currentlySigningIn = false;
+      signInTaskCompletionSource.setResult(null);
+    }
+  }
+
+  @Override
+  public void onActivityStarted(@NonNull @NotNull Activity activity) {
+    Log.d(TAG, "Started activity: " + activity.getClass().getName());
+  }
+
+  @Override
+  public void onActivityResumed(@NonNull @NotNull Activity activity) {
+    Log.d(TAG, "Resumed activity: " + activity.getClass().getName());
+
+    if (activity instanceof SignInResultActivity) {
+      return;
+    }
+    // throw error if app reentered during signin
+    if (currentlySigningIn) {
+      currentlySigningIn = false;
+      setSignInTaskCompletionError(new FirebaseAppDistributionException(AUTHENTICATION_FAILURE));
+    }
+    this.currentActivity = activity;
+  }
+
+  @Override
+  public void onActivityPaused(@NonNull @NotNull Activity activity) {
+    Log.d(TAG, "Paused activity: " + activity.getClass().getName());
+  }
+
+  @Override
+  public void onActivityStopped(@NonNull @NotNull Activity activity) {
+    Log.d(TAG, "Stopped activity: " + activity.getClass().getName());
+  }
+
+  @Override
+  public void onActivitySaveInstanceState(
+      @NonNull @NotNull Activity activity, @NonNull @NotNull Bundle bundle) {
+    Log.d(TAG, "Saved activity: " + activity.getClass().getName());
+  }
+
+  @Override
+  public void onActivityDestroyed(@NonNull @NotNull Activity activity) {
+    Log.d(TAG, "Destroyed activity: " + activity.getClass().getName());
```

Comment:
This comment feels a little incomplete. Either add more context or remove the comment entirely.